### DISCUSSION TITLE
Card creation via Regex?

### DISCUSSION THREAD

**Discussion by benhamilton** [roovo/obsidian-card-board#65] [Ideas]
Idea: use regex for column criteria

Currently on Tag based boards, cards are:
1. are markdown tasks
2. specified with a hashtag

For example: `- [ ] task details here #projecta/inprogress`

This creates two limits
1. The card must be a markdown tasks
2. We can only have a single hashtag per column

If we could use regex to specify what a column included, we could have a paragraph of text appear as a card i.e.
`A non task based card appears on the board #projecta/inprogess` (addresses limit 1)

Not only that but it also means that we could, via regex have more than one hashtag be in a single column i.e.
`This paragraph of details for a different project could appear in one master board #otherproject/inprogress` (addresses limit 2)

The benefit of using regex is that it would allow more freedom to specify what constitutes a card, and which column/s it should appear in.

For me personally, I want to have a board per project, but then one simplified board that simply has two columns that show what's to be done (i.e. Todo) and what's in progress (i.e. In Progress) across all projects and non-projects.

**Comment by roovo** [comment]
thanks!!

I'm trying to get an understanding of your use-cases... are there two here?  One being an ability to have non-tasks appearing on a board and another being the ability to have per project boards as well as a summary board?

For the second of these, if you are not wedded to the concept of having nested tags that define both a project and its status then the current bit of work I am doing (#22) might be a different way to achieve this. The idea would be that you could, for each board, define file, path, and tag filters that would apply.  That way you could limit one board to only tasks with tag #project1 and set up the columns as per your status tags and another board in the same way but with a different project tag.  The overview board wouldn't have a filter.  So you would have something like this for the project 1 board:

![Screenshot 2021-12-13 at 21 18 02](https://user-images.githubusercontent.com/5251/145890883-eeff7d00-df06-48d0-8ab9-c626eed8170d.png)

On the other point, are you wanting to combine task and non-task items on the same board or do you have something in mind for non-task related boards?

**Reply by benhamilton** [reply]
Yes, there are in fact two use cases in that - I should have really split that.

For the **first use case**, _having non-tasks appear as cards on the board_: yes, I'd like non-tasks, i.e. lines that do not contain `- [ ] ` appear as a card on a board. This would allow a *block* (line of markdown) appear as a card. This would not rely upon indentation or lack of indentation or the task prefix `- [ ] ` and it would mean that for writing up projects/proposals, one could simply tag a block i.e. `#wip` i.e. `.*\#.*(wip).*` or use org-mode style<sup>[1]</sup> convention `TODO ` i.e. `.*TODO.+` and have that block appear as a card on a board. This means it's easy to see which paragraph/block still requires attention. This opens up many, many other types of workflow without being restricted to having to have an unindented task to specify what appears. I do realise that using regex for this filter is somewhat of a power user feature... but it would be very useful to me.

For the **second use case**, _having a task appear in more than one board_:
#22 (and #27) will help with the second case. It would be nice if #22 at some point in the future supported multiple filters. i.e. `#tag1 AND #tag2` - this second use case can be accomplished if the first use case above is in play, because with regex you could specify the AND / OR etc. i.e. the regex<sup>[2]</sup> `\[.\]\s(.*)\#(pending|onhold)` would pickup all markdown tasks regardless of indent, that are pending or on hold<sup>[3]</sup>. This then means that I could use similar regex to put all _cancelled_ or _notdoing_ tasks in one column as well. 

For me, this second use case is more of a subset of use case one. If I had to choose... I'd be hopeful for use case one, the addition of regex filters.

---

Footnotes:
1. I know, not org-mode but similar
2. Note regex in Obsidian search seems to want the slashes in the other direction `/` rather than `\`
3. _pending_ means I'm waiting for a response from a client, _onhold_ means I'm waiting for a third party, i.e. the clients IT vendor, either way, I'm waiting on someone else
